driver/ixsystems/iscsi.py

```diff
@@ -281,10 +281,10 @@ def get_volume_stats(self, refresh=False):
     def create_cloned_volume(self, volume, src_vref):
         """Creates a volume from source volume."""
         LOG.info('iXsystems Create Cloned Volume')
-        LOG.info('create_cloned_volume: %s', src_vref['id'])
+        LOG.info('create_cloned_volume: %s', volume['id'])
 
         temp_snapshot = {'volume_name': src_vref['name'],
-                         'name': 'name-c%s' % src_vref['id']}
+                         'name': 'name-%s' % volume['id']}
 
```

Comment:
For https://github.com/iXsystems/cinder/issues/32, issue is caused by when create volume from volume, volume snapshot name hardcoded to parent volume name, which cause create second snapshot failed for the same parent volume. Fix change snapshot name to newly created volume id which is unique for each creation. Note: this should not be confused with openstack create snapshot from volume, openstack create snapshot from volume is separate action from create volume from volume.